repose-aggregator/components/filters/iri-validator/src/main/scala/org/openrepose/filters/irivalidator/IriValidatorFilter.scala

```diff
@@ -0,0 +1,42 @@
+package org.openrepose.filters.irivalidator
+
+import javax.servlet._
+import javax.servlet.http.{HttpServletRequest, HttpServletResponse}
+
+import com.typesafe.scalalogging.slf4j.LazyLogging
+import org.apache.jena.iri.{IRIException, IRIFactory}
+
+/**
+ * This filter validates that the request URI is a valid IRI.
+ */
+class IriValidatorFilter extends Filter with LazyLogging {
+
+  override def init(filterConfig: FilterConfig): Unit = {
+    logger.trace("IRI validator filter initialized")
+  }
+
+  override def destroy(): Unit = {
+    logger.trace("IRI validator filter destroyed")
+  }
+
+  override def doFilter(servletRequest: ServletRequest, servletResponse: ServletResponse, filterChain: FilterChain): Unit = {
+    val httpServletRequest = servletRequest.asInstanceOf[HttpServletRequest]
+    val httpServletResponse = servletResponse.asInstanceOf[HttpServletResponse]
+    val requestUrl = httpServletRequest.getRequestURL.toString + httpServletRequest.getQueryString
```

Comment:
I assumed we also wanted to validate the query string, but let me know if that is a bad assumption.